packages/devtools_app/lib/src/charts/chart_controller.dart

```diff
@@ -222,23 +228,43 @@ class ChartController extends DisposableController
     rightLabelTimestamp = null;
   }
 
-  // xCanvas coord for plotting data.
+  /// xCanvas coord for plotting data.
   double xCanvasChart;
 
-  // Width of the canvas for plotting data (#).
+  /// Width of the canvas for plotting data (#).
   double canvasChartWidth;
 
-  // Right-side padding after computing minPadding and max number of integral ticks for canvasChartWidth.
+  /// Right-side padding after computing minPadding and max number of integral ticks for canvasChartWidth.
   double xPaddingRight;
 
-  // yCanvas coord for plotting data.
+  /// yCanvas coord for plotting data.
   double yCanvasChart = 0;
 
-  // yCanvas height for plotting data.
+  /// yCanvas height for plotting data.
   double canvasChartHeight = 0;
 
-  // X axis ticks, each timestamp is a tick on the X axis.
-  final timestamps = <int>[];
+  bool get isDirty => dirty;
+
+  bool dirty = false;
+
+  // TODO(terry): Consider timestamps returning UnmodifiableListView
+  //              if loaded from a file (not live).
+  List<int> get timestamps => _timestamps;
+
+  /// X axis ticks, each timestamp is a tick on the X axis.
+  final _timestamps = <int>[];
+
+  void addTimestamp(int timestamp) {
+    _timestamps.add(timestamp);
+    dirty = true;
+  }
+
+  void timestampsClear() {
+    _timestamps.clear();
+    dirty = true;
+  }
+
+  int get timestampsSize => _timestamps.length;
```

Comment:
nit: why not call timestampsLength for consistency with dart core libs naming conventions?

Reply:
Done.